### PR TITLE
[PAN-2340] simple permissioning smart contract

### PR DESCRIPTION
**Pull request by macfarla**
## PR description
Simple smart contract for node permissioning - smart contract to add node, remove node, check if connection between nodes is allowed

## Fixed Issue(s)
fixes #PAN-2340

### CODE REVIEW

File: acceptance-tests/simple-permissioning-smart-contract/README.md [hunk 1]
```diff
@@ -0,0 +1,125 @@
+## Pre-requisites
+* [Truffle](https://truffleframework.com/) installed 
+```
+npm install -g truffle
+```
+* [Wallet](https://www.npmjs.com/package/truffle-hdwallet-provider) installed
+```
+npm install truffle-hdwallet-provider
+```
+## To run the pet shop example:
```
Comment (Errorific) [inline acceptance-tests/simple-permissioning-smart-contract/README.md]: pet shop

File: acceptance-tests/simple-permissioning-smart-contract/contracts/SimplePermissioning.sol [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+        EnodeIpv6 memory zeros = EnodeIpv6(bytes32(0), bytes32(0), bytes16(0), 0);
+        whitelistIpv6[key] = zeros;
+    }
+    function removeEnodeIpv4(bytes32 enodeHigh, bytes32 enodeLow, bytes4 enodeIpv4, uint16 enodePort) public {
+        bytes memory key = computeKeyIpv4(enodeHigh, enodeLow, enodeIpv4, enodePort);
+        EnodeIpv4 memory zeros = EnodeIpv4(bytes32(0), bytes32(0), bytes4(0), 0);
+        whitelistIpv4[key] = zeros;
+    }
+    function computeKeyIpv6(bytes32 enodeHigh, bytes32 enodeLow, bytes16 enodeIpv6, uint16 enodePort) public pure returns (bytes memory) {
+        //return abi.encodePacked(enodeHigh, enodeLow, enodeIpv4, enodePort);
```
Comment (Errorific) [inline acceptance-tests/simple-permissioning-smart-contract/contracts/SimplePermissioning.sol]: commented code

File: acceptance-tests/simple-permissioning-smart-contract/contracts/SimplePermissioning.sol [hunk 3]
```diff
@@ -0,0 +1,78 @@
...
+        return abi.encode(enodeHigh, enodeLow, enodeIpv6, enodePort);
+    }
+    function computeKeyIpv4(bytes32 enodeHigh, bytes32 enodeLow, bytes4 enodeIpv4, uint16 enodePort) public pure returns (bytes memory) {
+        //return abi.encodePacked(enodeHigh, enodeLow, enodeIpv4, enodePort);
```
Comment (Errorific) [inline acceptance-tests/simple-permissioning-smart-contract/contracts/SimplePermissioning.sol]: commented code

File: acceptance-tests/simple-permissioning-smart-contract/README.md (hunk 1 above)
Comment (macfarla) [reply acceptance-tests/simple-permissioning-smart-contract/README.md]: fixed

File: acceptance-tests/simple-permissioning-smart-contract/contracts/SimplePermissioning.sol (hunk 2 above)
Comment (macfarla) [reply acceptance-tests/simple-permissioning-smart-contract/contracts/SimplePermissioning.sol]: fixed

File: acceptance-tests/simple-permissioning-smart-contract/contracts/SimplePermissioning.sol (hunk 3 above)
Comment (macfarla) [reply acceptance-tests/simple-permissioning-smart-contract/contracts/SimplePermissioning.sol]: fixed

**Pull request merged**